From 5c3290520f6d53c089977965cdd92d0261696be7 Mon Sep 17 00:00:00 2001
From: Mihai Ene <mihai.ene@windriver.com>
Date: Tue, 12 Aug 2014 17:44:18 +0300
Subject: [PATCH 2/2] gpio: adapting Fintek f75111 gpio driver to work with
 aiot-x1000

Changed F75111_VER in order to match the chip on the aiot-x1000 board.
Modify f75111_get() to be aware of the direction of the gpio pin in order
to work also in "in" direction

Signed-off-by: Mihai Ene <mihai.ene@windriver.com>
---
 drivers/gpio/f75111.c  |    8 +++++++-
 include/linux/f75111.h |    2 +-
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/f75111.c b/drivers/gpio/f75111.c
index 25214ae..8c10402 100644
--- a/drivers/gpio/f75111.c
+++ b/drivers/gpio/f75111.c
@@ -257,7 +257,13 @@ static int f75111_get(struct gpio_chip *chip, unsigned offset)
 	struct f75111	*f75111 = container_of(chip, struct f75111, chip);
 	struct i2c_client *client = f75111->client;
 
-	return f75111_io(client, offset, F75111_GPIO_OUT_DATA, 0, 1);
+        /* get direction */
+        if (f75111_io(client, offset, F75111_GPIO_OUT_CTRL, 0, 1))
+                /* output */
+                return f75111_io(client, offset, F75111_GPIO_OUT_DATA, 0, 1);
+        else
+                /* input */
+                return f75111_io(client, offset, F75111_GPIO_INP_STATUS, 0, 1);
 }
 
 static int f75111_output(struct gpio_chip *chip, unsigned offset, int value)
diff --git a/include/linux/f75111.h b/include/linux/f75111.h
index b984545..9bfa963 100644
--- a/include/linux/f75111.h
+++ b/include/linux/f75111.h
@@ -66,7 +66,7 @@
 #define F75_VENDOR2			0x5E
 
 #define F75111_ID			0x0300
-#define F75111_VER			0x10
+#define F75111_VER			0x12
 #define FINTEC_ID			0x1934
 
 
-- 
1.7.5.4

